src/components/PaginationFooter/Buttons/Button.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import ArrowLeft from '@qlik-trial/sprout/icons/react/ArrowLeft';
+import ArrowLeftStop from '@qlik-trial/sprout/icons/react/ArrowLeftStop';
+import ArrowRight from '@qlik-trial/sprout/icons/react/ArrowRight';
+import ArrowRightStop from '@qlik-trial/sprout/icons/react/ArrowRightStop';
+import { StyledButton } from '../styles';
+import { DEFAULT_FONT_SIZE } from '../../../constants';
+import { usePaginationContext } from '../context/PaginationProvider';
+
+interface Props {
+  disabledCondition: boolean;
+  pageNumber: number;
+  type: string;
+  onKeyDown?: ((event: React.KeyboardEvent) => void) | null;
+}
+
+const ICONS: Record<string, typeof ArrowLeft> = {
+  FirstPage: ArrowLeftStop,
+  PreviousPage: ArrowLeft,
+  NextPage: ArrowRight,
+  LastPage: ArrowRightStop,
+  FirstPageRTL: ArrowRightStop,
+  PreviousPageRTL: ArrowRight,
+  NextPageRTL: ArrowLeft,
+  LastPageRTL: ArrowLeftStop,
+};
+
+const Button = ({ disabledCondition, pageNumber, type, onKeyDown }: Props) => {
```

Comment:
Since Button Component exists within the `src/components/PaginationFooter` it's more scoped with Pagination. 
So i think if you want to have more variants of this button (like `<FirstPage />`, `<LastPage />` and...) it's better to handle it within this button component and change it's name to some thing like `<PaginationButton />`. 

Or, you can take the other way and move Button component up into `src/components` and then create it's variants within `src/components/PaginationFooter` directory. That would be the best solution IMO, since it would give more freedom to Button component and make it more generic in case of other use cases

Reply:
Right now we don't have a need for a button anywhere else afaik. There is some logic in here that is very specific to the pagination, so I think it would just get bloated if we would cram more use-cases into this button. We could potentially rename the component, if that makes things clearer. As you mention, it is located in the paginationFooter folder, so it is pretty clear that it is a button made for the pagination

Reply:
I agree with @haxxmaxx on this one. It's not a generic button component. But rather an abstraction of common button logic for the pagination button.

Reply:
So in that case I would say let's have one generalised button for all 4 actions then let's have something like an array of pagination buttons data that takes care of all data of those 4 buttons: 
idealy something like below in a hight level component/jsx design: 

```jsx

// PaginationContext.tsx

const paginationBtns = [{...},{...},{...},...]

return (
  <PaginationProvider {...props}>
    <RowsPerPage />
    <SelectPage />
    <DisplayedRowsLabel />
    {paginationBtns.map(btn => <ThatGeneralizedButton {...btn} />)} // <- like this
  </PaginationProvider>
)

```

Reply:
I don't really see and need for a HOC here. I might as well just use a hook or just a simple function that resolves the common properties that is needed by `StyledButton`?

That would remove the need for the "generic" button and replace it with a single function that returns some props.

Reply:
seems good to me 👍